src/qibolab/transpilers/routing.py

```diff
@@ -0,0 +1,339 @@
+import matplotlib.pyplot as plt
+import networkx as nx
+import numpy as np
+from more_itertools import pairwise
+from qibo import gates
+from qibo.config import log, raise_error
+from qibo.models import Circuit
+
+from qibolab.transpilers.abstract import Transpiler, create_circuit_repr
+from qibolab.transpilers.placer import Trivial, assert_placement
+
+
+def respect_connectivity(connectivity, circuit, verbose=False):
+    """Checks if a circuit can be executed on Hardware.
+
+    Args:
+        circuit (qibo.models.Circuit): Circuit model to check.
+        connectivity (networkx.graph): chip connectivity.
+        verbose (bool): If ``True`` it prints info messages.
+
+    Returns ``True`` if the following conditions are satisfied:
+        - Circuit does not contain more than two-qubit gates.
+        - Circuit matches connectivity.
+    otherwise returns ``False``.
+    """
+
+    for gate in circuit.queue:
+        if len(gate.qubits) > 2 and not isinstance(gate, gates.M):
+            if verbose:
+                log.info(f"{gate.name} acts on more than two qubits.")
+            return False
+        elif len(gate.qubits) == 2:
```

Comment:
```suggestion
        if len(gate.qubits) == 2:
```
The code above returns, so this is not needed